src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,71 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.RegistrationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+    private static StorageDaoImpl storageDao;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @Test
+    void register_user_isOK() {
+        User expected = new User("BobGood", "1234567", 18);
+        registrationService.register(expected);
+        assertEquals(expected, storageDao.get(expected.getLogin()));
+    }
+
+    @Test
+    void register_userPasswordIsShort_isNotOK() {
+        User expected = new User("BobWithShortPassword", "12", 18);
+        assertThrows(RegistrationException.class, () -> registrationService.register(expected));
+    }
+
+    @Test
+    void register_userLoginIsShort_isNotOK() {
+        User expected = new User("Bob", "1234567", 18);
+        assertThrows(RegistrationException.class, () -> registrationService.register(expected));
+    }
+
+    @Test
+    void register_userUnderage_isNotOK() {
+        User expected = new User("BobChild", "1234567", 15);
+        assertThrows(RegistrationException.class, () -> registrationService.register(expected));
+    }
+
+    @Test
+    void register_userLoginIsNull_isNotOK() {
+        User expected = new User(null, "1234567", 18);
+        assertThrows(RegistrationException.class, () -> registrationService.register(expected));
+    }
+
+    @Test
+    void register_userPasswordIsNull_isNotOk() {
+        User expected = new User("BobWithoutPassword", null, 18);
+        assertThrows(RegistrationException.class, () -> registrationService.register(expected));
+    }
+
+    @Test
+    void register_userAgeIsNull_isNotOK() {
+        User expected = new User("BobWithoutAge", "1234567", null);
+        assertThrows(RegistrationException.class, () -> registrationService.register(expected));
+    }
+
+    @Test
+    void register_userAlreadyExists_isNotOK() {
+        User expected = new User("BobAlreadyExists", "john123", 28);
+        storageDao.add(expected);
+        assertThrows(RegistrationException.class, () -> registrationService.register(expected));
```

Comment:
In your `register_userAlreadyExists_isNotOK()` method, you are adding a user directly using the `storageDao.add()` method. This is not a good practice as it can cause your tests to be dependent on each other. Instead, you should add a method in your `StorageDaoImpl` class to clear the storage before each test.